app.ts

```diff
@@ -0,0 +1,301 @@
+import { ajax, css } from "jquery";
+class GlobalVariables {
+	firstNumber = 0;
+	lastNumber = 0;
+	backend = "http://localhost:2050";
+	resizeTimeout = 0;
+	//fetches the number of records from localhost
+	fetchRecordCount(): Promise<number> {
+		return fetch(`${this.backend}/recordCount`)
+			.then((res) => {
+				if (!res.ok) {
+					throw 'Failed to fetch record count';
+				}
+				return res.text();
+			})
+			.then((data) => {
+				return JSON.parse(data);
+			})
+			.catch((err) => {
+				throw 'Error fetching the record count: ' + err;
+			})
+	}
+	//fetches columns from localhost
+	fetchColumns(): Promise<string[]> {
+		return fetch(`${this.backend}/columns`)
+			.then(res => {
+				if (!res.ok) {
+					throw 'Failed to fetch columns';
+				}
+				return res.text();
+			})
+			.then((data) => {
+				return JSON.parse(data);
+			}).catch(err => {
+				throw 'Error fetching columns' + err;
+			})
+	}
+	//fetches records from localhost
+	fetchRecords(from: number, to: number): Promise<any[]> {
+		return fetch(`${this.backend}/records?from=${from}&to=${to}`)
+			.then(res => {
+				if (!res.ok) {
+					throw "Sorry, there's a problem with the network";
+				}
+				return res.text();
+			})
+			.then((data) => {
+				return JSON.parse(data);
+			}).catch(err => {
+				throw 'Error fetching records from server ' + err;
+			})
+	}
+}
+
+const globalVars = new GlobalVariables();
+let recordCount: number;
+
+globalVars.fetchRecordCount()
+	.then((count) => {
+		// Assign the fetched count to the variable
+		recordCount = count - 1;
+		displayRecords(recordCount);
+	})
+	.catch(err => {
+		throw 'Error fetching record count' + err;
+	})
+//Initializes the table head
+function createTable() {
+	return globalVars.fetchColumns()
+		.then(columns => {
+			for (const col of columns) {
+				$(".head").append(`<th>${col}</th>`);
+			}
+		})
+		.catch(err => {
+			throw 'Error creating table' + err;
+		})
+}
+//calculates the number of rows that can fit the screen
+const adjustRowsByScreenHeight = (): number => {
```

Comment:
This function seems to return something, but the name of the function sounds like it is doing something instead, please rename it to something more fitting.